app.py

```diff
@@ -1,26 +1,116 @@
-from modules.Preprocessing import get_tweet_data
-from modules.Sentiment_Analysis import train_model_if_necessary, analyze_many_tweets
-
-# Train model
-train_model_if_necessary()
-# scrape here
-print('Scraping...')
-tweet_data = [{}]  # do stuff
-# open file for preprocessing
-print('Preprocessing...')
-# FILENAME = './'
-# with open(FILENAME) as file_object:
-#     tweet_data = file_object.read()
-
-handled_tweets = get_tweet_data(tweet_data)  # do stuff
-
-# analysis here
-print('Performing sentiment analysis...')
-analyzed_tweets = *handled_tweets  # do stuff
-
-# presentation here
-print('Generating presentations...')
-result = *analyzed_tweets  # do stuff
+from modules.web_scraper import get_tweets
+from modules.Preprocessing import handle_tweet_data
+# from modules.Preprocessing import handle_tweet_data
+# from modules.Sentiment_Analysis import train_model_if_necessary, analyze_many_tweets
+# import modules.presentation as presentation
+import argparse
+
+
+#########CUSTOM TYPES#########
+def _restricted_float(val: float):
+    try:
+        val = float(val)
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            f"{val} not a floating-point literal")
+
+    if 0.0 < x > 1.0:
+        raise argparse.ArgumentTypeError(f"{val} not in range [0.0, 1.0]")
+    return val
```

Comment:
What happens here?

Reply:
So, on an argument you can pass a `type` such as `float`, which we need here. But it doesn't provide any checking besides if it can convert the value to float.

We also need to check if the value is within a range og floats ([0.0-1.0]) so it is easier to move the type check into its own function - and it is also the recommended way to handle types in an argparser.

On top of that we get custom error messages. Argparse' error messages are generally good enough, though.

I will be implementing custom types like this for every argument.